pkg/tidb/forwarder.go

```diff
@@ -118,6 +115,28 @@ func (f *Forwarder) pollingForTiDB() {
 	}
 }
 
+func (f *Forwarder) resolveSqlAddr(override string) (string, error) {
+	if override != "" {
+		return override, nil
+	}
+	if f.sqlProxy.noAliveRemote.Load() {
+		log.Warn("Unable to resolve sql connection address since no alive TiDB instance")
+		return "", ErrNoAliveTiDB.NewWithNoMessage()
+	}
+	return fmt.Sprintf("127.0.0.1:%d", f.sqlPort), nil
```

Comment:
These two functions seems to be too verbose. Is it possible to reduce some duplicates? For example, I believe it can be as simple as:

```golang
func (f *Forwarder) getEndpointAddr(port int) (string, error) {
    if f.sqlProxy.noAliveRemote.Load() {
        return "", ErrNoAliveTiDB.NewWithNoMessage()
    }
    return fmt.Sprintf("127.0.0.1:%d", port), nil
}
```

Note: I exclude the "override" logic because this is NOT a behaviour that a forwarder / proxyer / load balancer should have.